src/Sarif.Multitool/Rules/RuleResources.resx

```diff
@@ -308,13 +312,15 @@ Many tool use similar names for 'uriBaseId' symbols. We suggest 'REPOROOT' for t
     <value>{0}: Placeholder</value>
   </data>
   <data name="SARIF1012_MessageArgumentsMustBeConsistentWithRule_Error_MessageIdMustExist_Text" xml:space="preserve">
-    <value>{0}: Placeholder '{1}' '{2}'</value>
+    <value>{0}: This message object refers to the message with id '{1}' in rule '{2}', but that rule does not define a message with that id. When a tool creates a result message that uses the 'id' property, it must ensure that the specified rule actually has a message with that id.</value>
   </data>
   <data name="SARIF1012_MessageArgumentsMustBeConsistentWithRule_Error_SupplyEnoughMessageArguments_Text" xml:space="preserve">
-    <value>{0}: Placeholder '{1}' '{2}' '{3}' '{4}' '{5}'</value>
+    <value>{0}: The message with id '{1}' in rule '{2}' requires {3} arguments, but the 'arguments' array in this message object has only {4} elements. When a tool creates a result message that use the 'id' and 'arguments' properties, it must ensure that the 'arguments' array has enough elements to provide values for every replacement sequence in the message specified by 'id'. For example, if the highest numbered replacement sequence in the specified message string is '{{3}}', then the 'arguments' array must contain 4 elements.</value>
   </data>
   <data name="SARIF1012_MessageArgumentsMustBeConsistentWithRule_FullDescription_Text" xml:space="preserve">
-    <value>Placeholder</value>
+    <value>The properties of a result's 'message' property must be consistent with the properties of the rule that the result refers to.
+
+When a result's 'message' object uses the 'id' and 'arguments' properties (which, by the way, is recommended: see SARIF), it must ensure that the rule actually defines a message string with that id, and that 'arguments' array has enough elements to provide values for every replacement sequence in the message specified by 'id'. For example, if the highest numbered replacement sequence in the specified message string is '{3}', then the 'arguments' array must contain at least 4 elements.</value>
```

Comment:
>SARIF [](start = 114, length = 5)

The link to the section is missing. Was it not in the document?

Reply:
yes, this is what i see in md, let me cross check if i have latest version


---
In reply to: [449255220](https://github.com/microsoft/sarif-sdk/pull/1960#discussion_r449255220) [](ancestors = 449255220)

Reply:
Yes, my mistake, stand by for fix.

---
In reply to: [449255884](https://github.com/microsoft/sarif-sdk/pull/1960#discussion_r449255884) [](ancestors = 449255884,449255220)